crypto/evp/evp_enc.c

```diff
@@ -192,6 +197,7 @@ static int evp_cipher_init_internal(EVP_CIPHER_CTX *ctx,
         ctx->fetched_cipher = (EVP_CIPHER *)cipher;
     }
     ctx->cipher = cipher;
+    ctx->key_len = -1;
```

Comment:
Likewise, shouldn't this only happen on `key != NULL`?

Reply:
Same as above?